dart/lib/src/sentry_client.dart

```diff
@@ -366,7 +361,44 @@ class SentryClient {
 
   void close() => _options.httpClient.close();
 
-  Future<SentryEvent?> _processEvent(
+  Future<SentryEvent?> _runBeforeSend(
+    SentryEvent event, {
+    Hint? hint,
+  }) async {
+    SentryEvent? eventOrTransaction = event;
+
+    final beforeSend = _options.beforeSend;
+    final beforeSendTransaction = _options.beforeSendTransaction;
+    String beforeSendName = 'beforeSend';
+
+    try {
+      if (event is SentryTransaction && beforeSendTransaction != null) {
+        beforeSendName = 'beforeSendTransaction';
+        eventOrTransaction = await beforeSendTransaction(event);
+      } else if (beforeSend != null) {
+        eventOrTransaction = await beforeSend(event, hint: hint);
+      }
+    } catch (exception, stackTrace) {
+      _options.logger(
+        SentryLevel.error,
+        'The $beforeSendName callback threw an exception',
+        exception: exception,
+        stackTrace: stackTrace,
+      );
+    }
+
+    if (eventOrTransaction == null) {
+      _recordLostEvent(event, DiscardReason.beforeSend);
```

Comment:
We need to check if there's a discard reason specifically for the transaction.

Reply:
Looked over at `sentry-javascript`, the initial PR implementing this feature always used `beforeSend` as reason. We should probably check if the server knows about `beforeSendTransaction`?

Reply:
Just confirmed, we should keep using `beforeSend` for now.

Reply:
thx! 🙇